src/foundry/foundry.js/collections/documentCollections/compendiumCollection.d.ts

```diff
@@ -0,0 +1,216 @@
+
+    /** Track whether the Compendium Collection is private */
+    get private(): boolean;
+
+    /** A convenience reference to the label which should be used as the title for the Compendium pack. */
+    get title(): string;
+
+    get(key: string, { strict }: { strict: true }): DocumentInstanceForMetadata<T>;
+    get(key: string, { strict }?: { strict?: false }): DocumentInstanceForMetadata<T> | undefined;
+
+    set(id: string, document: DocumentInstanceForMetadata<T>): this;
+
+    delete(id: string): boolean;
+
+    /** Load the Compendium index and cache it as the keys and values of the Collection. */
+    getIndex(): Promise<this['index']>;
+
+    /**
+     * Get a single Document from this Compendium by ID.
+     * The document may already be locally cached, otherwise it is retrieved from the server.
+     * @param id -  The requested Document id
+     * @returns The retrieved Document instance
+     */
+    getDocument(id: string): Promise<DocumentInstanceForMetadata<T> | undefined>;
+
+    /**
+     * Load multiple documents from the Compendium pack using a provided query object.
+     * @param query - A database query used to retrieve documents from the underlying database
+     *                default: `{}`
+     * @returns The retrieved Document instances
+     */
+    getDocuments(query?: Query): Promise<DocumentInstanceForMetadata<T>[]>;
+
+    /**
+     * Import a Document into this Compendium Collection.
+     * @param document - The existing Document you wish to import
+     * @returns The imported Document instance
+     */
+    importDocument(document: DocumentInstanceForMetadata<T>): Promise<DocumentInstanceForMetadata<T>>;
+
+    /**
+     * Fully import the contents of a Compendium pack into a World folder.
+     * @returns The imported Documents, now existing within the World
+     */
+    importAll({ folderId, folderName, options }: Partial<ImportAllOptions>): Promise<DocumentInstanceForMetadata<T>[]>;
+
+    /**
+     * Add a Document to the index, capturing it's relevant index attributes
+     * @param document -The document to index
+     */
+    indexDocument(document: DocumentInstanceForMetadata<T>): void;
+
+    /**
+     * Create a new Compendium Collection using provided metadata.
+     * @param metadata - The compendium metadata used to create the new pack
+     * @param options - Additional options which modify the Compendium creation request
+     *                  default `{}`
+     */
+    static createCompendium<T extends Metadata<any>>(
+      metadata: T,
+      options?: Partial<DocumentModificationOptions>
+    ): Promise<CompendiumCollection<T>>;
+
+    /**
+     * Assign configuration metadata settings to the compendium pack
+     * @param settings - The object of compendium settings to define
+     *                   default: `{}`
+     * @returns A Promise which resolves once the setting is updated
+     */
+    configure(settings?: Partial<CompendiumConfiguration>): Promise<void>;
```

Comment:
I think the return type should be `Promise<CompendiumConfiguration>` as `ClientSettings#set` returns a `Promise` of the thing that has been passed in.